support-services/src/main/scala/com/gu/support/redemption/DynamoTableAsync.scala

```diff
@@ -0,0 +1,82 @@
+package com.gu.support.redemption
+
+import java.util.concurrent.{Future => JFuture}
+
+import com.amazonaws.AmazonWebServiceRequest
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.dynamodbv2.model._
+import com.amazonaws.services.dynamodbv2.{AmazonDynamoDBAsync, AmazonDynamoDBAsyncClient}
+import com.gu.support.promotions.dynamo.DynamoService
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+object DynamoTableAsync {
+  def apply(table: String, key: String)(implicit e: ExecutionContext): DynamoTableAsync = {
+    val dynamoDBClient = AmazonDynamoDBAsyncClient.asyncBuilder
+      .withCredentials(DynamoService.CredentialsProvider)
+      .withRegion(Regions.EU_WEST_1)
+      .build()
+    new DynamoTableAsync(dynamoDBClient, table, key)
+  }
+}
+trait DynamoLookup {
+  def lookup(value: String): Future[Option[Map[String, AttributeValue]]]
+}
+
+trait DynamoUpdate {
+  def update(primaryKeyValue: String, updateName: String, updateValue: Boolean): Future[Unit]
+}
```

Comment:
the two traits are to prevent the operation having to depend on the whole service, which is good in itself.  The practical benefit of that is easier reuse in the tests, the test mocks can just be a simple inline single abstract method with a pattern match, which makes the tests a hundred times clearer and banishes mockito 🎉

Reply:
perhaps a better option would be to separate the below `DynamoTableAsync` into two classes.  I put them together to reduce the amount of stuff we have and because they both depend on the same parameters, but in hindsight that seems a bit unnecessary to have them grouped

Reply:
Fine to leave them grouped I think

Reply:
will do, but interesting comment, I find it easier to use code with fewer interdependencies, which is actually not the same as fewer classes, as things often have to be finer grained to avoid a dependency that's not strictly necessary.  I wonder if there's some element of tradeoff there.